src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs

```diff
@@ -46,30 +45,25 @@ internal class UnifiedSuggestedActionsSource
             // it. However, it's deliberate.  We want to make sure that the code runs on 
             // the background so that no one takes an accidentally dependency on running on 
             // the UI thread.
-            var fixes = Task.Run(
+            var fixes = await Task.Run(
                 () => codeFixService.GetFixesAsync(
                 document, selection, includeSuppressionFixes, isBlocking,
-                addOperationScope, cancellationToken), cancellationToken).WaitAndGetResult(cancellationToken);
+                addOperationScope, cancellationToken), cancellationToken).ConfigureAwait(false);
 
-            var filteredFixes = FilterOnUIThread(fixes, workspace);
+            var filteredFixes = FilterOnAnyThread(fixes);
             var organizedFixes = OrganizeFixes(workspace, filteredFixes, includeSuppressionFixes);
 
             return organizedFixes;
         }
 
-        private static ImmutableArray<CodeFixCollection> FilterOnUIThread(
-            ImmutableArray<CodeFixCollection> collections,
-            Workspace workspace)
+        private static ImmutableArray<CodeFixCollection> FilterOnAnyThread(ImmutableArray<CodeFixCollection> collections)
```

Comment:
should this just be named Filter (or similar)?  seems like the current pattern is only mention in the method name if it requires the UI thread, otherwise it can be assumed the ui thread is not needed?

Reply:
Yes, the `OnAnyThread` suffix is unnecessary. But it's also a private method so either way is fine.